tools/scylla-nodetool.cc

```diff
@@ -239,6 +239,25 @@ void statusgossip_operation(scylla_rest_client& client, const bpo::variables_map
     fmt::print(std::cout, "{}\n", status.GetBool() ? "running" : "not running");
 }
 
+void stop_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    if (vm.count("id")) {
+        throw std::invalid_argument("stopping compactions by id is not implemented");
+    }
+    if (!vm.count("compaction_type")) {
+        throw std::invalid_argument("missing required parameter: compaction_type");
+    }
+
+    static const std::vector<std::string_view> recognized_compaction_types{"COMPACTION", "CLEANUP", "SCRUB", "RESHAPE", "RESHARD", "UPGRADE"};
+
+    const auto compaction_type = vm["compaction_type"].as<sstring>();
+
+    if (std::ranges::find(recognized_compaction_types, compaction_type) == recognized_compaction_types.end()) {
+        throw std::invalid_argument(fmt::format("invalid compaction type: {}", compaction_type));
```

Comment:
avoiding comment about std::ranges::contains

Reply:
I'm very much looking forward to `std::ranges::contains()`, I don't understand what took them so long to introduce it.

Reply:
They always keep something good for the next version to get us to upgrade.

Reply:
Especially "contains" was always one version behind the "find" function ;-)